airflow/models/taskinstance.py

```diff
@@ -3731,6 +3814,12 @@ def from_ti(cls, ti: TaskInstance) -> SimpleTaskInstance:
             key=ti.key,
             run_as_user=ti.run_as_user if hasattr(ti, "run_as_user") else None,
             priority_weight=ti.priority_weight if hasattr(ti, "priority_weight") else None,
+            # Inspect the ti, to check if the 'dag_run' relationship is loaded.
+            parent_context_carrier=ti.dag_run.context_carrier
+            if "dag_run" not in inspect(ti).unloaded
+            else None,
+            context_carrier=ti.context_carrier if hasattr(ti, "context_carrier") else None,
+            span_status=ti.span_status if hasattr(ti, "span_status") else None,
```

Comment:
`ti` will always have this attribute won't it?

Reply:
Yes, you are right. There is a default value of `NOT_STARTED`.

Reply:
I'll simplify it.